File: mailer/src/router.js

import { Router } from "express"
import { authenticate } from "./middlewares.js"
import { contactFromAdmin, contactFromUser, newAccountMessageController, resetPasswordController, updateEmailController } from "./controllers.js"

export const router = Router()

router.post("/api/mailer/contact-test", authenticate("ROLE"), (req, res) => {
    res.status(200).send("Contact test successful")
})

router.post("/api/mailer/email-change", updateEmailController)
router.post("/api/mailer/reset-password", resetPasswordController)

router.post("/api/mailer/messages/from-admin", contactFromAdmin)
router.post("/api/mailer/messages/from-user", contactFromUser)

router.post("/api/mailer/new-account", newAccountMessageController)

File: mailer-rs/src/router.rs

use axum::Router;
use axum::routing::post;

use crate::controllers::{email_change, reset_password, contact_from_admin, contact_from_user, new_account_message, new_instance_message};

// Function to create and return a router with email-related routes
pub fn mailer_router() -> Router {

    Router::new()

        .route("/email-change", post(email_change)

        )
        .route("/reset-password", post(reset_password)

        )
        .route("/messages/from-admin", post(contact_from_admin)

        )
        .route("/messages/from-user", post(contact_from_user)

        )
        .route("/new-account", post(new_account_message)

        )                
        .route("/new-instance", post(new_instance_message)

        )                
}
